cpmpy/transformations/linearize.py

```diff
@@ -550,7 +558,8 @@ def canonical_comparison(lst_of_expr):
 
         elif isinstance(cpm_expr, Comparison):
             lhs, rhs = cpm_expr.args
-            if isinstance(lhs, Comparison) and cpm_expr.name == "==":  # reification of comparison
+            if isinstance(lhs, Comparison) and (is_bool(rhs) or isinstance(rhs, Expression) and rhs.is_bool()):
+                assert cpm_expr.name == "==", "Expected a reification of a comparison here, but got {}".format(cpm_expr.name)
```

Comment:
so if the comparison has bool on both side, we expect this to be a full reification. I'm a bit worried about inputs of linear constraints having PB terms?

- Is `var bool >= var bool` handled by this function?
- What if we give this `var bool = var bool`? We see this as a reified constraint as well, or a linear to be canonicalized? I guess we depend on this having the same outcome?